src/main/deltachat/backup.ts

```diff
@@ -14,8 +14,29 @@ const log = getLogger('main/deltachat/backup')
 import SplitOut from './splitout'
 import { DeltaChatAccount } from '../../shared/shared-types'
 export default class DCBackup extends SplitOut {
-  export(dir: string) {
-    this._dc.importExport(C.DC_IMEX_EXPORT_BACKUP, dir, undefined)
+  async export(dir: string) {
+    this._dc.stopIO()
+    try {
+      await this._internal_export(dir)
+    } catch (err) {
+      this._dc.startIO()
+      throw err
+    } finally {
+      this._dc.startIO()
+    }
+  }
+
+  private async _internal_export(dir: string) {
+    return new Promise((resolve, reject) => {
+      this._dc.importExport(C.DC_IMEX_EXPORT_BACKUP, dir, undefined)
+      this._dc.on('DC_EVENT_IMEX_PROGRESS', (data1: number) => {
+        if (data1 === 0) {
+          reject('Backup export failed (progress==0)')
+        } else if (data1 === 1000) {
+          resolve()
+        }
```

Comment:
Please clean up the event listener inside this function

Reply:
right that was forgotten here